docs/sdks/uid2-sdk-ref-java.md

```diff
@@ -102,6 +103,183 @@ Available information returned through the SDK is outlined in the following tabl
 | `KeysNotSynced` | The client has failed to synchronize keys from the UID2 service. |
 | `VersionNotSupported` |  The client library does not support the version of the encrypted token. |
 
+-------------------------------------------copy begin-----------------------------------------------
+
+Copy from: https://github.com/IABTechLab/uid2-client-java#usage-for-publishers
+
+## Usage for Publishers
+
+As a publisher, there are two ways to use the UID2 SDK for Java: 
+1. [**Basic Usage**](#basic-usage) is for publishers who want to use this SDK's HTTP implementation (synchronous [OkHttp](https://square.github.io/okhttp/)).
+2. [**Advanced Usage**](#advanced-usage) is for publishers who prefer to use their own HTTP library. 
+
+For an example application that demonstrates both Basic and Advanced usage, see [Java UID2 Integration Example](https://github.com/UnifiedID2/uid2-examples/tree/main/publisher/uid2-java-test-site#readme).
+
+### Basic Usage
+
+If you're using the SDK's HTTP implementation, follow these steps.
+
+1. Create an instance of `PublisherUid2Client` as an instance variable:
+
+   ```java
+   private final PublisherUid2Client publisherUid2Client = new PublisherUid2Client(UID2_BASE_URL, UID2_API_KEY, UID2_SECRET_KEY);
+   ```
+
+2. Call a function that takes the user's email address or phone number as input and generates a `TokenGenerateResponse` object. The following example uses an email address:
+   ```java
+   TokenGenerateResponse tokenGenerateResponse = publisherUid2Client.generateTokenResponse(TokenGenerateInput.fromEmail(emailAddress).doNotGenerateTokensForOptedOut());
+   ```
+
+   :::important
+   <ul><li>Be sure to call this function only when you have obtained legal basis to convert the user’s [directly identifying information (DII)](../ref-info/glossary-uid.md#gl-dii) to UID2 tokens for targeted advertising.</li></ul>
+   
+   <ul><li>Always apply `doNotGenerateTokensForOptedOut()`. This applies `policy=1` in the [/token/generate](../endpoints/post-token-generate.md#token-generation-policy) call. Support for `policy=0` will be removed soon. (**GWH_SW01 to adjust this here + in the repo pending input. Readme line 53**)</li></ul>
```

Comment:
_Support for `policy=0` will be removed soon._  can be omitted.

The rest can stay as is

Reply:
Maybe tweak this to:

_Always apply doNotGenerateTokensForOptedOut(). This applies a param similar to setting optout_check=1 in the [/token/generate](http://localhost:3006/docs/endpoints/post-token-generate#token-generation-policy) call._

policy=1 is not documented on the endpoint doc any more so we best not to say it. But i am trying to say doNOtGenerateTokensForOptedOut will apply policy=1 which has same effect as optout_check=1 so reader don't need to care about underlying implementation detail but just call the function. But maybe add something like:

_Not calling this doNOtGenerateTokensForOptedOut method is only available for early UID2 participants._ 

(and don't even mention policy=0 as they probably dunno what that is)

Applies this to comment further below too.

Reply:
@sunnywu I fixed it to this (2 instances here, 2 in the repo readme):

IMPORTANT: Always apply doNotGenerateTokensForOptedOut(). This applies a parameter similar to setting optout_check=1 in the call to the POST /token/generate endpoint (see [Unencrypted JSON Body Parameters](https://unifiedid.com/docs/endpoints/post-token-generate#unencrypted-json-body-parameters)).

I'd prefer to leave out the other part about early UID2 participants. Would you be OK with that? If you feel strongly that we should include it let me know. Thx!

Reply:
I'm ok either way but if the early participants change something and they wanna update the vesion of sdk, they need to know to call doNotGenerateTokensForOptedOut() hence my rationale. Yeh maybe don't mention it for now.

Reply:
@Sunny thank you. Merging.